crates/hir-def/src/hir.rs

```diff
@@ -255,6 +255,7 @@ pub enum Expr {
         path: Option<Box<Path>>,
         fields: Box<[RecordLitField]>,
         spread: Option<ExprId>,
+        ellipsis: bool,
```

Comment:
Can we encode this with a ternary enum? Like
```
enum Spread {
    No,
    Yes,
    Base(ExprId)
}
```

Reason being that `ellipsis: false` `spread: Some` is a bogus state.

Name ideas (for variants/enum) welcome